### DISCUSSION TITLE
How to pass where, order_by to $db.queryRaw same as prisma.entity.findMany?

### DISCUSSION THREAD

**Discussion by michael-land** [prisma/prisma#3746] [Q&A] [answered]
Is there any helper function we can use to pass where, order_by parameters to `db.queryRaw`?

I have a function that returns sets of table, I want pass `order_by` and `where` same as prisma client generated api. I can't use a VIEW here because i need to pass extra parameters to my functions.

e.g. 
```sql
create function report.get_statistic("startDate" date, "endDate" date) returns SETOF report.statistic
    stable
    language sql
as
$$
-- FUNCTION DEFINITION
$$
```

```ts
await db.$queryRaw('SELECT * FROM report.get_statistic(${from}, ${to})')`;
```

Is there any way to do it like this

```ts
await db.$queryRaw(`
SELECT *
FROM report.get_statistic(${from}, ${to})
WHERE ${transformWhere(where)}
ORDER BY ${transformOrderBy(orderBy)}`)`
```

**Comment by ryands17** [comment]
@xiaoyu-tamu 👋 

Could you try something like this:

```ts
await db.$queryRaw(`
SELECT *
FROM report.get_statistic($1, $2)
WHERE ${transformWhere(where)}
ORDER BY ${transformOrderBy(orderBy)}`,
from, to)
```

`db.$queryRaw` is the same as any raw query so passing where and order by will work as in any normal raw query.

**Comment by michael-land** [comment, marked as answer]
@ryands17 

Hi, thanks for quick reply. Does prisma export helper functions `transformWhere` and `transformOrderBy`? 

My`where` and `order by` input are exact same shape as the following doc described.

 https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/filtering#filter-conditions

**Reply by ryands17** [reply]
No, that unfortunately is not supported. You would need to write these transformations by yourself.

**Reply by ryands17** [reply]
It would be great if you could create a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) explaining your use case.

**Reply by michael-land** [reply]
> It would be great if you could create a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) explaining your use case.

Will do. 

Thanks, just want to make sure before i implement it myself :-)

**Reply by mhuggins** [reply]
@michael-land Did you open a feature request for this?  I need the same thing.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
